hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/util/CompareData.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.engine.util;
+
+import org.hawkular.alerts.api.model.data.Data;
+
+/**
+ * CompareConditions often work against data arriving at different times. We convert Data into CompareData and
+ * apply special handling such that CompareData can live longer than one engine firing.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+public class CompareData {
```

Comment:
Right, I actually cherry-picked the PR for master from this branch, where I found the problem...  So this should disappear from the PR after another rebase, I think.  Even without a rebase it will not be a conflict.